### ISSUE TITLE
YAML language support

### ISSUE THREAD

**Issue by vojtechhabarta** [Microsoft/monaco-languages#2]
I would like to have YAML support in Monaco Editor. Ideally it should support syntax highlighting and provide extensible completion lists (maybe based on some schema language).

I found that YAML syntax highlighting in VS Code is implemented [here](https://github.com/Microsoft/vscode/blob/master/extensions/yaml/syntaxes/YAML.plist) but this is in TextMate format which cannot be used in Monaco Editor (reason is mentioned [here](https://github.com/Microsoft/vscode/issues/174#issuecomment-158083640)).

Is there any possibility to convert TextMate grammar to Monarch grammar?
Are there any plans for YAML support?

**Comment by alexandrudima** [comment]
@vojtechhabarta You can very easily jump in and create a YAML colorizer. https://microsoft.github.io/monaco-editor/monarch.html is a playground page that allows you to write rules in the left hand side editor and preview them live applied on the right hand side editor. You could then make a PR against this repository to add it such that everybody gets it with the monaco-editor.

Here is for example a commit that moved markdown over to this repo (it nicely shows what things to edit to add a new language): https://github.com/Microsoft/monaco-languages/commit/23b305079f10cb365b8badda1a57d263a31a6b41)

TL;DR; PR welcome! :heart:

**Comment by jrieken** [comment]
This issue was moved to Microsoft/monaco-editor#131
